cmake/modules/dts.cmake

```diff
@@ -292,7 +292,7 @@ execute_process(
   RESULT_VARIABLE ret
   ERROR_VARIABLE stderr
   )
-if(NOT "${ret}" STREQUAL "0")
+if(NOT "${ret}" STREQUAL "0" OR stderr)
```

Comment:
perhaps better to take a step back and remove the patching of paching by simply allow CMake to handle this situation and to ensure CMake always prints both stdout and stderror.

Thus simplify the code to:
```
execute_process(
  COMMAND ${CMD_GEN_DEFINES}
  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
  COMMAND_ERROR_IS_FATAL ANY
)
```

and then completely remove the logic of
```
if(NOT "${ret}" STREQUAL "0" OR stderr)
  if (stderr)
    ...
  else()
    ..
  endif()
else()
```

That will ensure:
- Warnings from gen_defines are printed on console
- Errors in gen_defines will fail CMake and print the error.

And if we then extend `gen_defines` to take `--warnings-as-errors` flag, then it's clearer where the responsibility of dts warnings / errors belongs, and CMake can then apply the flag when the devicetree code has been cleaned up and thereby failing the build on dts warnings (the `--warning-as-errors` can be done as independent PR).